frontend/model/contracts/group.js

```diff
@@ -672,6 +721,8 @@ DefineContract({
               Vue.set(groupProfile, key, value)
           }
         }
+        const income = data.incomeDetailsType === 'incomeAmount' ? data.incomeAmount - state.settings.mincomeAmount : data.pledgeAmount
```

Comment:
Why are you subtracting the mincome from their income? Did you mean to add the mincome to the pledge?

Reply:
Apparently answering by email puts the github replies into the wrong discussion thread. As replied (in the wrong thread), here is my answer:

VERY good question!

I *kind of* (but not entirely) answer this at the very beginning of the
as-of-incomplete mocha tests:

```
// Please note: "income"s are the user pledges if positive, OR the needed
income amount ABOVE mincome if NEGATIVE.
```

(Negative values above a value are below).

From a CS/math perspective: this approach uses the 'sign-bit' of our
floating-point values to preserve the information about whether they are
pledging (positive) or needing (negative).

These positive/negative values are how above/below the mincome a
haver/needer is.

I should mention: this calculation is entirely reversible, and will NOT
affect the heuristics of our group-consolidated approach.

I assure you: this encoding is compatible with both the strictly-pro-rated
solution, as well as the group-consolidated version (because it is
reversible).

Also:instead of passing the minCome as a time-dependant-parameter to our
algorithm, it needs to be modeled as an event (including upon group
creation). This is because the minCome can change at any time during a
monthly cycle.

I should also mention: this above/below bit-encoding-of-the-sign-bit as an
indication of having/needing is a feature for making the pro-weighted
approach less computationally complex, and more immediately obvious as a
heuristic approach.

Reply:
OK, then since this isn't actually their income, let's rename this parameter everywhere it appears. Both here, in the function name, and in the data that is set in the event.

This isn't an income, it's something else. It is a pledge or a need. Or a have/need. So maybe let's call it `haveOrNeed`.

Reply:
Will change this in next push.